### PR TITLE
Add Testimonial and Stack components

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Adds new UI patterns for `Testimonial` per the [specification in Figma](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=1455%3A30880). 

Also adds the bones for a new `Stack` layout helper component, inspired by [this Primer design specification](https://github.com/github/primer/issues/1078) (Hubbers only 🔒). 

## List of notable changes:

- **added** new components
- **added** Doctocat docs for Testimonial
- **added** Storybook stories and playgrounds for `Testimonial` **and** `Stack`
- **updated** Primer Primitives dependencies to `@latest`

## What should reviewers focus on?

- Try out the Playgrounds for [Stack](https://primer-37f4c6bed0-26139705.drafts.github.io/storybook/?path=/story/components-stack--playground) and [Testimonial](https://primer-37f4c6bed0-26139705.drafts.github.io/storybook/?path=/story/components-testimonial--playground)
- [Read the docs for Testimonial](https://primer-37f4c6bed0-26139705.drafts.github.io/components/Testimonial)
- Sanity check the naming for props and compound component children
- Code review 

## Supporting resources (related issues, external links, etc):

- [Figma](https://www.figma.com/file/BJ95AjraesmRCWsKA013GS/Primer-Brand?node-id=1455%3A30880)
- [Stack](https://github.com/github/primer/issues/1078)
- [MUI Stack](https://mui.com/material-ui/react-stack/)

## Contributor checklist:

- [x] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### `Testimonial`
Two testimonial components in horizontal layout- both showing quotes from David Ross who says - GitHub helps us ensure that we have our security controls baked into our pipelines all the way from the first line of code we are writing

#### Storybook playground for Testimonial

https://user-images.githubusercontent.com/13340707/192296814-b607bd68-e1db-4173-aaa3-ab30fe1dd05a.mov

### `Stack`
https://user-images.githubusercontent.com/13340707/192295690-6433dad1-9d39-4005-93e8-326a0867f629.mov

### CODE REVIEW

File: docs/content/components/Testimonial.mdx [hunk 1]
```diff
@@ -0,0 +1,164 @@
+---
+title: Testimonial
```
Comment (colebemis) [inline docs/content/components/Testimonial.mdx:2]: Should Stack also have a documentation page?

File: src/Stack/Stack.module.css [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+
+.Stack-narrow--vertical {
+  flex-direction: column;
+}
+
+.Stack-narrow--horizontal {
+  flex-direction: row;
+}
+
+@media screen and (min-width: 48rem) {
```
Comment (colebemis) [inline src/Stack/Stack.module.css:24]: Can we use design tokens in @media queries?

**Review by colebemis — APPROVED** [review]
Love the Stack component 🔥 Nice work on this!

File: src/Stack/Stack.tsx [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  /**
+   * Defines the gap CSS property.
+   * A string value will be applied to all viewports.
+   * An object can be used to define different values for different viewports.
+   */
+  gap?: StackSpacingVariants | ResponsiveSpacingMap
+  /**
+   * Applies vertical alignment to child elements relative to the Stack, using the align-items CSS property.
+   */
+  alignItems?: 'center' | 'flex-start' | 'flex-end'
```
[the remark this answers is not in the document]
Comment (rezrah) [reply src/Stack/Stack.tsx:42]: that's a good point.. i'm not sure 🤔 ... it does make sense..i'll add it in 👍

File: docs/content/components/Testimonial.mdx (hunk 1 above)
Comment (rezrah) [reply docs/content/components/Testimonial.mdx:2]: Yeah, let me add one in. I was originally keeping it as an internal helper component, with examples in Storybook but makes sense to have it in main docs too 👍

File: src/Stack/Stack.module.css (hunk 2 above)
Comment (rezrah) [reply src/Stack/Stack.module.css:24]: Not natively, but spoke to @langermank yesterday about this and there's a `postcss` processor that can do some compile-time substitution using our viewport tokens, which we do have available.

@langermank - are you able to share please? Alternatively, happy to put our own up. Either way, I'll likely put it in a separate PR and do a sweep of all other hardcoded @media queries in Primer Brand.

**Pull request merged**